modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -226,6 +226,7 @@
   protected ExecutionMediator executionMediator;
   protected CompletableComponentExecutor componentExecutor;
   protected ReturnDelegate returnDelegate;
+  protected ReturnDelegate valueReturnDelegate;
```

Comment:
why do we need this instead of simply reusing the same returnDelegate? What's the difference?

Reply:
this deserves a large javadoc explaining why

Reply:
male it private, not protected